internal/venice-common/src/main/java/com/linkedin/venice/pubsub/api/PubSubProducerAdapterContext.java

```diff
@@ -0,0 +1,166 @@
+  }
+
+  public PubSubMessageSerializer getPubSubMessageSerializer() {
+    return pubSubMessageSerializer;
+  }
+
+  public boolean isProducerCompressionEnabled() {
+    return isProducerCompressionEnabled;
+  }
+
+  public String getCompressionType() {
+    return compressionType;
+  }
+
+  public static class Builder {
+    private String producerName;
+    private String brokerAddress;
+    private VeniceProperties veniceProperties;
+    private PubSubSecurityProtocol securityProtocol;
+    private MetricsRepository metricsRepository;
+    private PubSubTopicRepository pubSubTopicRepository;
+    private PubSubMessageSerializer pubSubMessageSerializer;
+    private boolean shouldValidateProducerConfigStrictly = true;
+    private boolean isProducerCompressionEnabled = true;
+    private String compressionType;
+
+    public Builder setProducerName(String producerName) {
+      this.producerName = producerName;
+      return this;
+    }
+
+    public Builder setBrokerAddress(String brokerAddress) {
+      this.brokerAddress = brokerAddress;
+      return this;
+    }
+
+    public Builder setVeniceProperties(VeniceProperties veniceProperties) {
+      this.veniceProperties = veniceProperties;
+      return this;
+    }
+
+    public Builder setSecurityProtocol(PubSubSecurityProtocol securityProtocol) {
+      this.securityProtocol = securityProtocol;
+      return this;
+    }
+
+    public Builder setMetricsRepository(MetricsRepository metricsRepository) {
+      this.metricsRepository = metricsRepository;
+      return this;
+    }
+
+    public Builder setPubSubTopicRepository(PubSubTopicRepository pubSubTopicRepository) {
+      this.pubSubTopicRepository = pubSubTopicRepository;
+      return this;
+    }
+
+    public Builder setShouldValidateProducerConfigStrictly(boolean shouldValidateProducerConfigStrictly) {
+      this.shouldValidateProducerConfigStrictly = shouldValidateProducerConfigStrictly;
+      return this;
+    }
+
+    public Builder setPubSubMessageSerializer(PubSubMessageSerializer pubSubMessageSerializer) {
+      this.pubSubMessageSerializer = pubSubMessageSerializer;
+      return this;
+    }
+
+    public Builder setProducerCompressionEnabled(boolean isProducerCompressionEnabled) {
+      this.isProducerCompressionEnabled = isProducerCompressionEnabled;
+      return this;
+    }
+
+    public Builder setCompressionType(String compressionType) {
+      this.compressionType = compressionType;
+      return this;
+    }
+
+    public PubSubProducerAdapterContext build() {
+      if (brokerAddress == null) {
+        throw new VeniceException("Broker address must be provided to create a pub-sub producer");
+      }
+      if (veniceProperties == null) {
+        veniceProperties = VeniceProperties.empty();
+      }
+      if (pubSubMessageSerializer == null) {
+        pubSubMessageSerializer = PubSubMessageSerializer.DEFAULT_PUBSUB_SERIALIZER;
+      }
+      if (compressionType == null) {
```

Comment:
wait... is this right?  Couldn't it be NONE?

Reply:
I've changed this to make it straightforward as per you suggestion. Thanks!